src/hotspot/share/logging/logDecorations.cpp

```diff
@@ -25,117 +25,114 @@
 #include "jvm.h"
 #include "logging/logConfiguration.hpp"
 #include "logging/logDecorations.hpp"
-#include "runtime/atomic.hpp"
 #include "runtime/os.hpp"
 #include "runtime/thread.inline.hpp"
 #include "services/management.hpp"
 
-const char* volatile LogDecorations::_host_name = NULL;
-
-LogDecorations::LogDecorations(LogLevelType level, const LogTagSet &tagset, const LogDecorators &decorators)
-    : _level(level), _tagset(tagset) {
-  create_decorations(decorators);
-}
-
-const char* LogDecorations::host_name() {
-  const char* host_name = Atomic::load_acquire(&_host_name);
-  if (host_name == NULL) {
+// We cache the hostname at startup under the assumption that it seldom changes
+class CachedHostName {
+  const char* const _v;
+  static const char* get_host_name() {
     char buffer[1024];
     if (os::get_host_name(buffer, sizeof(buffer))) {
-      host_name = os::strdup_check_oom(buffer);
-      const char* old_value = Atomic::cmpxchg(&_host_name, (const char*)NULL, host_name);
-      if (old_value != NULL) {
-        os::free((void *) host_name);
-        host_name = old_value;
-      }
+      // Use raw malloc here
+      return ::strdup(buffer);
```

Comment:
Hi Xin, I think you review an old version, I reverted that coding.

But the point about using a pointer and resolving hostname lazily on first access is to prevent paying anything if logging is off or logging is on but the hostname decorator is not used. For a static char array you always have to pay.

Reply:
Sorry. forget it.   I commented yesterday and just sent out. 
I see you revert host_name(). The reason to use lazy initialization is fair enough.